overlord/snapstate/snapstate.go

```diff
@@ -1360,6 +1360,90 @@ func installWithDeviceContext(st *state.State, name string, opts *RevisionOption
 	return doInstall(st, &snapst, snapsup, 0, fromChange, nil)
 }
 
+func Download(ctx context.Context, st *state.State, name string, blobDirectory string, opts *RevisionOptions, userID int, flags Flags, deviceCtx DeviceContext) (*state.TaskSet, error) {
+	if opts == nil {
+		opts = &RevisionOptions{}
+	}
+
+	if opts.CohortKey != "" && !opts.Revision.Unset() {
+		return nil, errors.New("cannot specify revision and cohort")
+	}
+
+	if opts.Channel == "" {
+		opts.Channel = "stable"
+	}
+
+	var snapst SnapState
+	err := Get(st, name, &snapst)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return nil, err
+	}
+
+	if err := snap.ValidateInstanceName(name); err != nil {
+		return nil, fmt.Errorf("invalid instance name: %v", err)
+	}
+
+	sar, err := downloadInfo(ctx, st, name, opts, userID, deviceCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	info := sar.Info
+
+	if blobDirectory == "" {
+		blobDirectory = dirs.SnapBlobDir
+	}
```

Comment:
this is not necessary as it's taken care again elsewhere?

Reply:
I made a change that addresses this.